projects/igniteui-angular/src/lib/grids/grid/grid-filtering.spec.ts

```diff
@@ -1084,6 +1084,23 @@ describe('IgxGrid - Filtering expression tree bindings #grid', () => {
         expect(grid.rowList.length).toEqual(8);
         expect(grid.filteringExpressionsTree.filteringOperands.length).toEqual(0);
     }));
+
+    it('should correctly set filteredData if advancedFilteringExpressionsTree is empty', () => {
+        const tree = new FilteringExpressionsTree(FilteringLogic.And);
+        tree.filteringOperands = [];
+        grid.filteringExpressionsTree.filteringOperands = [];
+        grid.filteringExpressionsTree.type = 1;
+
+        // Clear filter
+        grid.advancedFilteringExpressionsTree = tree;
+        fix.detectChanges();
+
+        // Verify filtering expressions tree binding state
+        expect(grid.advancedFilteringExpressionsTree).toEqual(fix.componentInstance.filterTree);
```

Comment:
Why dont compare against `tree`, instead of `fix.componentInstance.filterTree` ?

Reply:
Did this in my new commit 👍